docs/frontend-system/architecture/07-routes.md

```diff
@@ -8,143 +8,420 @@ description: Frontend routes
+
+Another thing to note is that this indirection in the routing is particularly useful for open source plugins that need to leave flexibility in how they are integrated. For plugins that you build internally for your own Backstage application, you can choose to go the route of direct imports or even use concrete routes directly. Although there can be some benefits to using the full routing system even in internal plugins. It can help you structure your routes, and as you will see further down it also helps you manage route parameters.parameters.
+
+### Optional External Route References
+
+When creating an ExternalRouteRef it is possible to mark it as optional:
+
+```tsx
+// plugins/catalog/src/routes.ts
+import {
+  createRouteRef,
+  createExternalRouteRef,
+} from '@backstage/frontend-plugin-api';
+
+const rootRouteRef = createRouteRef();
+const createComponentRouteRef = createExternalRouteRef({
+  optional: true,
+});
+```
+
+An external route that is marked as optional is not required to be bound in the app, allowing it to be used as a switch for whether a particular link should be displayed or action should be taken.
+
+When calling useRouteRef with an optional external route, its return signature is changed to RouteFunc | undefined, allowing for logic like this:
+
+```tsx
+// plugins/catalog/src/plugin.tsx
+import { createPlugin, createPageExtension, useRouteRef } from '@backstage/frontend-plugin-api';
+import { rootRouteRef, createComponentRouteRef } from './routes';
+
+const catalogEntityListPage = createPageExtension({
+  defaultPath: '/'
+  routeRef: rootRouteRef,
+  loader: async () => {
+    const href = useRouteRef(catalogCreateComponentRouteRef)();
+
+    return (
+      <div>
+        <h1>Catalog Entities</h1>
+        {/* Since the route is optional, rendering the link only if the href is defined */}
+        {href && <a to={href}>Create Component</a>
+      </div>
+    );
+  }
+});
+
+export default const createPlugin({
+  id: 'catalog',
+  routes: {
+    entityList: rootRouteRef
+  }
+  externalRoutes: {
+    createComponent: createComponentRouteRef,
+  },
+  extensions: [catalogEntityListPage]
+});
+
+// index.ts
+export { default } from './plugin';
+```
+
+## Sub Route References
+
+The last kind of route refs that can be created are `SubRouteRefs`, which can be used to create a route ref with a fixed path relative to an absolute `RouteRef`. They are useful if you have a page that internally is mounted at a sub route of a page extension component, and you want other plugins to be able to route to that page.
+
+For example:
+
+```tsx
+// plugins/catalog/src/routes.ts
+import {
+  createRouteRef,
+  createSubRouteRef
+} from '@backstage/frontend-plugin-api';
+
+const rootRouteRef = createRouteRef();
+const detailsRouteRef = createSubRouteRef({
+  parent: rootRouteRef,
+  path: '/details',
+});
```

Comment:
Perhaps we can weave another example SubRouteRef with params straight into this example? Since parameterized sub routes are a little bit different from the rest

Reply:
Done 🙌🏻